thunder/executors/cudnnex.py

```diff
@@ -464,19 +480,53 @@ def cudnn_sdpa_backward_meta(
     philox_offset: TensorLike,
     *,
     scale: None | float = None,
-) -> (TensorProxy, TensorProxy, TensorProxy):
-    grad_query = TensorProxy(like=query)
-    grad_key = TensorProxy(like=key)
-    grad_value = TensorProxy(like=value)
+    cat_grad_qkv: bool,
+) -> tuple[TensorProxy, ...]:
+    if cat_grad_qkv:
+        grad_qkv = TensorProxy(
```

Comment:
The compile flag cudnn_sdpa_bwd_may_cat_grad_qkv suggests that cudnnex "may" output concatenated grads. Seems like its just assumed that cudnnex always be able to concatenate grads, as this code path only returns `grad_qkv`.
(Relying on cudnn to be able to do output grads in a strided fashion is a safe assumption for future cudnn versions too, as long as the embedding dim stride is 1.)

Reply:
> Seems like its just assumed that cudnnex always be able to concatenate grads

No, cat_grad_qkv is not the same as may_cat_grad_qkv. It combines `may_cat_grad_qkv` and [some size/stride checks](https://github.com/Lightning-AI/lightning-thunder/pull/59/files#diff-783705c9be7fba72357ae0678dac1da2961ddb64d37a40eb6e3bf03558f8e313R681) indicating "is able to cat".